### ISSUE TITLE
Access to export via command line

### ISSUE THREAD

**Issue by mshiels** [SubtitleEdit/subtitleedit#5559]
I know export can be done if you have loaded an image based subtitle, but wondering if that is accessible at all from command line.  Basically I want to do a bit of idx/sub and sup conversions,  for my automated post processor for videos.

**Comment by niksedk** [comment]
You can try something like `subtitleedit /convert a.sub srt` - it will use the last settings used in the UI I think... 
You can specify ocr engine: `ocrengine:tesseract` or `ocrengine:nocr`

**Comment by mshiels** [comment]
I was hoping to go image format to image format. 

 

Most other tools that do image subtitle formats, are getting old/buggy etc.  

if it’s not doable I may go code spelunking to see how hard it would be.

**Comment by niksedk** [comment]
OK, what are you trying to do precisely? VobSub to ?

**Comment by mshiels** [comment]
VobSub to VobSub to split some of them  when they have multi language and VObSub to PSG/Sup, PSG/Sup to VobSub.

**Comment by mshiels** [comment]
So niksedk something it can do? could easily be modified to do? or you are already updating it todo?  If it's not something you are working on I may hopin and see what I can do.

**Comment by mshiels** [comment]
OK, after checking out the code ya it already does image to image - perfect.  But for the life of me I can't get it to work. 

subtitleedit /convert zero.sub vobsub /inputfolder:Z:\ /outputfolder:Z:\tmp

still ends up OCRing things, and the way I read the code it should bypass the OCR.   So confused.

**Comment by niksedk** [comment]
Hm, VObSub to PSG/Sup will IMO definitively require OCR as images will look very poor otherwise (as vobsub has low quality).

SE only has a few image-to-image methods implemented, like TS to PGS, or PGS to BdnXml.

**Comment by mshiels** [comment]
Well the VObSubs in question are generated it seems fo 1920x1080 at least specified as such, so thinking they were generated instead of PSG/Sup.  but I'll see if I can find other tools, thx.

**Comment by mshiels** [comment]
AND I found a very strange situation that I need to deal with and FFMpeg is not being very friendly about it. 

I have a idx/sub that has it looks like 10 languages - I want to split it out ONLY the english into a sub/idx and embed just that subtitle, seems there are lots of muxers, but no demuxers.  I am thinking not hard given the engines capabilities??  

Literally I expected FFMpeg to report a idx/sub with 10 streams, but for some reason it only reports as stream 0x20 and 0x29 for 2:1, and 2:2, strange.   

Basically I am working on a pretty massive script that I run to get subtitles onto most of my tv stuff, but it does everything from a matching file, to demux/ccextractor, and now adding SubTitleEDit conversion of pgs/sub to text as well.  Basically future proofing I hope all the stuff I am watching, srt being pretty universal.  But I still mux in the pgs/sub - except these multi language ones are driving me slowly crazy!!!!  

Right now I have run int a sub/idx that has 4 English subtitle 'indexes' so I was trying to find a way to turn them into individual sub/idx since I found that works ok with ffmpeg - but combined - they just don't work I swear.   None are properly labelled but I was working on number of entries etc, to label one as forced, one as SDH etc.   Except I can't figure out how to demux it into individual files, then manipulate them as needed. 

But having tried the command line of SUbTitleEdit for some ocr to srt, it's pretty slick! hats off on a tricky conversion, but it works pretty flawlessly.  

PS: I just accidentally figured out part of my problem, refer to IDX file instead of SUB and sure enough all the streams appear, wow, in all my searching for FFmpeg idx/sub handling usually I find they refer to the .sub I thought.  live and learn!!!!
